scripts/download-sponsors.js

```diff
@@ -2,64 +2,190 @@
-        twitterHandle: node.fromAccount.twitterHandle,
-        description: node.fromAccount.description,
-      }))
-  )
-  .then(sponsors => JSON.stringify(sponsors, null, 2))
-  .then(sponsorsJson => {
-    fs.writeFile(sponsorsFile, sponsorsJson, err => {
-      if (err) {
-        console.error("Failed to write website/data/sponsors.json file: ", err);
-      } else {
-        console.log("Success: website/data/sponsors.json created.");
+  let allNodes = [];
+
+  // Handling pagination if necessary
+  // eslint-disable-next-line
+  while (true) {
+    const result = await fetch(graphqlEndpoint, {
+      method: "POST",
+      body: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    }).then(response => response.json());
+    const nodes = getNodes(result.data);
+    allNodes = [...allNodes, ...nodes];
+    body.variables.offset += graphqlPageSize;
+    if (nodes.length < graphqlPageSize) {
+      return allNodes;
+    } else {
+      // sleep for a while
+      await new Promise(resolve => setTimeout(resolve, 100));
+    }
+  }
+};
+
+const uniqBy = (arr, predicate) => {
+  const cb = typeof predicate === "function" ? predicate : o => o[predicate];
+
+  return [
+    ...arr
+      .reduce((map, item) => {
+        const key = item === null || item === undefined ? item : cb(item);
+
+        map.has(key) || map.set(key, item);
+
+        return map;
+      }, new Map())
+      .values(),
+  ];
+};
+
+(async () => {
+  const members = await getAllNodes(
+    membersGraphqlQuery,
+    data => data.account.members.nodes
+  );
+  let supporters = members
+    .map(nodeToSupporter)
+    .sort((a, b) => b.totalDonations - a.totalDonations);
+
+  // Deduplicating supporters with multiple orders
+  supporters = uniqBy(supporters, "slug");
+
+  const supportersBySlug = new Map();
+  for (const supporter of supporters) {
+    for (const key of REQUIRED_KEYS) {
+      if (!supporter || typeof supporter !== "object") {
+        throw new Error(
+          `Supporters: ${JSON.stringify(supporter)} is not an object.`
+        );
       }
-    });
-  })
-  .catch(err => {
-    console.error("Failed to fetch backers: ", err);
-  });
+      if (!(key in supporter)) {
+        throw new Error(
+          `Supporters: ${JSON.stringify(supporter)} doesn't include ${key}.`
+        );
+      }
+    }
+    supportersBySlug.set(supporter.slug, supporter);
+  }
+
+  // Calculate monthly amount from transactions
+  const transactionsYear = await getAllNodes(
+    transactionsGraphqlQuery,
+    data => data.transactions.nodes
+  );
+  for (const transaction of transactionsYear) {
+    if (!transaction.amountInHostCurrency) continue;
+    const amount = transaction.amountInHostCurrency.value;
+    if (!amount || amount <= 0) continue;
+    const supporter = supportersBySlug.get(transaction.fromAccount.slug);
+    if (!supporter) continue;
+    supporter.yearlyDonations += amount;
```

Comment:
Not related: I wished decimal proposal were stage 4.

Reply:
If it causes problems we can switch it to use `1` = `1 cent` rather than `1$`